config_gen.py

```diff
@@ -202,6 +202,25 @@ def run(cmd, *args, **kwargs):
         print("$ " + " ".join(cmd))
         subprocess.call(cmd, *args, **kwargs)
 
+    # helper function to set CC and CXX variables correctly on OSX with qmake
+    def patch_qmake_makefile(makefile_path):
+        compilers = {
+            "CC": "clang",
+            "CXX": "clang++",
+            "LINK": "clang++"
+        }
+        patched_makefile_lines = []
+        with open(makefile_path, "r") as f:
```

Comment:
This could be expressed more concisely by applying the `sub()` to the entire file at once, rather than line by line. (The check on `match()` is redundant.)